control-plane/agents/src/bin/core/pool/operations_helper.rs

```diff
@@ -130,3 +138,73 @@ impl OperationGuardArc<ReplicaSpec> {
         }
     }
 }
+
+pub(crate) async fn reject_same_devlink(
+    request: &CreatePool,
+    node: Arc<RwLock<NodeWrapper>>,
+    registry: &Registry,
+) -> Result<(), SvcError> {
+    let request_disks: HashSet<String> = request
+        .disks
+        .iter()
+        .map(|disk| utils::disk::normalize_disk(disk.as_str()))
```

Comment:
this won't work with pcie unbound devices?

Reply:
Have made the changes accordingly.